org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLHighlighting.java

```diff
@@ -73,20 +111,27 @@ private static boolean covers(Range range, Position position) {
 		return isBeforeOrEqual(range.getStart(), position) && isBeforeOrEqual(position, range.getEnd());
 	}
 
-	private static Range getTagNameRange(TokenType tokenType, XMLDocument xmlDocument, int startOffset) {
-		Scanner scanner = XMLScanner.createScanner(xmlDocument.getText(), startOffset);
+
+	private static Range getTagNameRange(TokenType tokenType, int startOffset,
+																				XMLDocument fmDocument) {
+
+		Scanner scanner = XMLScanner.createScanner(fmDocument.getText(), startOffset);
+
 		TokenType token = scanner.scan();
 		while (token != TokenType.EOS && token != tokenType) {
 			token = scanner.scan();
 		}
 		if (token != TokenType.EOS) {
 			try {
-				return new Range(xmlDocument.positionAt(scanner.getTokenOffset()),
-						xmlDocument.positionAt(scanner.getTokenEnd()));
+				return new Range(fmDocument.positionAt(scanner.getTokenOffset()),

```

Comment:
why fmDocument? It should be xmlDocument

Reply:
you're right, I don't know why I even did that.

Reply:
@NikolasKomonen you sjould make sure unit tests catch that